src/langcheck/metrics/en/query_based_text_quality.py

```diff
@@ -226,3 +226,135 @@ def harmful_activity(
+    or 1.0 (Good). The score may also be `None` if it could not be computed.
+
+    We currently only support the evaluation based on an EvalClient.
+
+    Args:
+        generated_outputs: The model generated output(s) to evaluate
+        prompts: The prompts used to generate the output(s)
+        eval_model: The EvalClient instance used for the evaluation
+
+    Returns:
+        A :class:`~langcheck.metrics.metric_value.MetricValue` object
+    """
+    metric_inputs = get_metric_inputs(
+        generated_outputs=generated_outputs,
+        prompts=prompts,
+        required_params=["generated_outputs", "prompts"],
+    )
+    metric_name = "summarization_quality"
+
+    summarization_quality_template = eval_model.load_prompt_template(
+        language=LANG, metric_name=metric_name
+    )
+
+    return eval_model.compute_metric_values_from_template(
+        metric_inputs=metric_inputs,
+        template=summarization_quality_template,
+        metric_name=metric_name,
+        language=LANG,
+        score_map={"0": 0.0, "0.5": 0.5, "1": 1.0},
+    )
+
+
+def system_prompt_adherence(
+    generated_outputs: list[str] | str,
+    prompts: list[str] | str,
+    system_prompts: list[str] | str,
+    eval_model: EvalClient,
+) -> MetricValue[float | None]:
+    """Calculates how well the system prompt is followed when the LLM generates
+    outputs. This metric takes on float values of either 0.0 (Not Adherent),
+    0.5 (Partially Adherent), or 1.0 (Adherent). The score may also be `None` if
+    it could not be computed.
+
+    We currently only support the evaluation based on an EvalClient.
+
+    Args:
+        generated_outputs: The model generated output(s) to evaluate
+        prompts: The prompts used to generate the output(s)
+        system_prompts: The system prompts used to generate the output(s)
+        eval_model: The EvalClient instance used for the evaluation
+
+    Returns:
+        A :class:`~langcheck.metrics.metric_value.MetricValue` object
+    """
+    metric_inputs = get_metric_inputs(
+        generated_outputs=generated_outputs,
+        prompts=prompts,
+        additional_inputs={
+            "system_prompts": system_prompts,
+        },
+        additional_input_name_to_prompt_var_mapping={
+            "system_prompts": "system_prompt",
+        },
+        required_params=["generated_outputs", "prompts", "system_prompts"],
+    )
+    metric_name = "system_prompt_adherence"
+
+    system_prompt_adherence_template = eval_model.load_prompt_template(
+        language=LANG, metric_name=metric_name
+    )
+
+    return eval_model.compute_metric_values_from_template(
+        metric_inputs=metric_inputs,
+        template=system_prompt_adherence_template,
+        metric_name=metric_name,
+        language=LANG,
+        score_map={"0": 0.0, "0.5": 0.5, "1": 1.0},
+    )
+
+
+def user_frustration(
+    prompts: list[str] | str,
+    chat_history: list[str] | str,
```

Comment:
Could you make it `history` in the argument name? It matches our internal naming convention better, and there could be some items that are not necessarily "chat history" but you still want to provide them to the metric (e.g. tool use history??)

Reply:
Makes sense, fixed! [0b23eeb](https://github.com/citadel-ai/langcheck/pull/197/commits/0b23eebaeeeb1278d724bb4103c859422de043d2)